jwst/outlier_detection/outlier_detection_step.py

```diff
@@ -96,11 +95,18 @@ def _build_reffile_container(self, reftype):
         reffile_to_model = {'gain': datamodels.GainModel,
             'readnoise': datamodels.ReadnoiseModel}
 
-        reffiles = [self.get_reference_file(im, reftype) for im in self.input_models]
+        if reftype in self.input_models[0].meta.ref_file._instance:
+            reffile_names = [tm.meta.ref_file._instance[reftype]['name'] for tm in self.input_models]
+            if len(set(reffile_names)) == 1:
+               reffiles = [self.get_reference_file(self.input_models[0],reftype)]*len(reffile_names)
+            else:
+                reffiles = [self.get_reference_file(im, reftype) for im in self.input_models]                 
```

Comment:
I looked at doing that, but the filename stored in the metadata comes wrapped in the URL for CRDS which needs to be interpreted before it can be used.  The current implementation results in a filename that can be used directly, unlike the value in the metadata.